pages/api/applications/[applicationId]/tasks/index.ts

```diff
@@ -1,5 +1,128 @@
-  res.status(200).end(`Request sent to ${req.url}\n`);
+const prisma = new PrismaClient();
+
+enum MessageType {
+  EMPTY_TITLE,
+  INVALID_DUE_DATE,
+  INVALID_NOTIFICATION_DATETIME,
+  TASK_APPLICATION_NOT_FOUND,
+  TASK_APPLICATION_DOES_NOT_BELONG_TO_USER,
+  TASK_CREATE_SUCCESSFULLY,
+}
+
+const messages = new Map<MessageType, StatusMessage[]>([
+  [MessageType.EMPTY_TITLE, [{ type: StatusMessageType.Error, message: 'Task title is empty.' }]],
+  [MessageType.INVALID_DUE_DATE, [{ type: StatusMessageType.Error, message: 'Task due date is invalid.' }]],
+  [
+    MessageType.INVALID_NOTIFICATION_DATETIME,
+    [{ type: StatusMessageType.Error, message: 'Task notification date and time is invalid.' }],
+  ],
+  [
+    MessageType.TASK_APPLICATION_NOT_FOUND,
+    [{ type: StatusMessageType.Error, message: 'Application for this task cannot be found.' }],
+  ],
+  [
+    MessageType.TASK_APPLICATION_DOES_NOT_BELONG_TO_USER,
+    [{ type: StatusMessageType.Error, message: 'Application for this task does not belong to you.' }],
+  ],
+  [MessageType.TASK_CREATE_SUCCESSFULLY, [{ type: StatusMessageType.Success, message: 'Task created succesfully.' }]],
+]);
+
+function handler(userId: string, req: NextApiRequest, res: NextApiResponse) {
+  const method = req.method;
+
+  switch (method) {
+    case HTTP_POST_METHOD:
+      handlePost(userId, req, res);
+      break;
+    default:
+      rejectHttpMethod(res, method);
+  }
 }
+
+async function handlePost(
+  userId: string,
+  req: NextApiRequest,
+  res: NextApiResponse<ApiResponse<TaskData | EmptyPayload>>,
+) {
+  if (!isValidRequest(req, res)) {
+    return;
+  }
+
+  const taskPostData: TaskPostData = {
+    ...req.body,
+    dueDate: new Date(req.body.dueDate),
+    notificationDateTime: req.body.notificationDateTime ? new Date(req.body.notificationDateTime) : null,
+  };
+
+  const applicationId = Number(req.query.applicationId);
+  const application = await prisma.application.findFirst({ where: { id: applicationId } });
+
+  if (!application) {
+    res
+      .status(HTTP_STATUS_NOT_FOUND)
+      .json(createJsonResponse({}, messages.get(MessageType.TASK_APPLICATION_NOT_FOUND)));
+    return;
+  }
+
+  if (application.userId !== userId) {
+    res
+      .status(HTTP_STATUS_FORBIDDEN)
+      .json(createJsonResponse({}, messages.get(MessageType.TASK_APPLICATION_DOES_NOT_BELONG_TO_USER)));
+    return;
+  }
+
+  createIfPossible(res, async () => {
+    const newTask = await prisma.task.create({
+      data: {
+        applicationId,
+        ...taskPostData,
+      },
+      select: { id: true, title: true, dueDate: true, notificationDateTime: true, isDone: true },
+    });
+
+    res
+      .status(HTTP_STATUS_CREATED)
+      .json(createJsonResponse(newTask, messages.get(MessageType.TASK_CREATE_SUCCESSFULLY)));
+  });
+}
+
+function isValidRequest(req: NextApiRequest, res: NextApiResponse<ApiResponse<EmptyPayload>>): boolean {
```

Comment:
`isValidRequest` implies it only checks and returns bool. Its also better practice to handle the res in the handler function (for clarity). Could we return the messageType if it fails or undefined if it passes, then use it in the handler?


```suggestion
function validateRequestOrReject(req: NextApiRequest, res: NextApiResponse<ApiResponse<EmptyPayload>>): MessageType | undefined {
```